tools/toolchains/arm.py

```diff
@@ -98,17 +98,16 @@ def version_check(self):
         stdout, _, retcode = run_cmd([self.cc[0], "--vsn"], redirect=True)
         msg = None
         min_ver, max_ver = self.ARMCC_RANGE
-        first_line = stdout.splitlines()[0]
-        match = self.ARMCC_VERSION_RE.match(first_line)
-        if match:
-            found_version = LooseVersion(match.group(1))
-            if found_version < min_ver or found_version > max_ver:
-                msg = ("Compiler version mismatch: Have {}; "
-                       "expected >= {} < {}"
-                       .format(found_version, min_ver, max_ver))
-        else:
+        match = self.ARMCC_VERSION_RE.search(stdout)
+        found_version = LooseVersion(match.group(0)) if match else None
+        min_ver, max_ver = self.ARM_RANGE
+        if found_version and (found_version < min_ver or found_version >= max_ver):
+            msg = ("Compiler version mismatch: Have {}; "
+                   "expected version >= {} and < {}"
+                   .format(found_version, min_ver, max_ver))
+        elif len(match.groups()) != 1:
```

Comment:
match can be None. dereferencing it will cause different error

Reply:
please ignore. you have the check

Reply:
Actually, I think this is correct.